src/test/java/io/javalin/openapi/TestOpenApi.kt

```diff
@@ -397,4 +421,13 @@ class TestOpenApi {
             assertThat(actualHeaders.getFirst("Access-Control-Allow-Methods")).isEqualTo("GET")
         }
     }
+
+    @Test
+    fun `composed schema via dsl`() {
+        val app = Javalin.create {
+            it.registerPlugin(OpenApiPlugin(OpenApiOptions(::simpleAnyOfSchema)))
+        }
+        val actual = JavalinOpenApi.createSchema(app)
+        assertThat(actual.asJsonString()).isEqualTo(dslComposedSchema)
+    }
```

Comment:
I think you misunderstood me. I didn't meant the cc.vileda DSL. I meant the DSL you can use with `document()`. Check the first test for examples.

I would expect an API like the following:

```kotlin
val anyOfRouteDocumentation = document()
        .result("200", documentedContent(anyOf(User::class, Address::class)))
```

Reply:
I prepared something:

In `DocumentedContent.kt` add the following below the `DocumentedContent` class:

```
class OneOf(val classes: Array<Class<*>>)
fun oneOf(vararg classes: Class<*>) = OneOf(classes.toList().toTypedArray())
fun oneOf(vararg classes: KClass<*>) = OneOf(classes.map { it.java }.toTypedArray())

class AnyOf(val classes: Array<Class<*>>)
fun anyOf(vararg classes: Class<*>) = AnyOf(classes.toList().toTypedArray())
fun anyOf(vararg classes: KClass<*>) = AnyOf(classes.map { it.java }.toTypedArray())

class AllOf(val classes: Array<Class<*>>)
fun allOf(vararg classes: Class<*>) = AllOf(classes.toList().toTypedArray())
fun allOf(vararg classes: KClass<*>) = AllOf(classes.map { it.java }.toTypedArray())
```

and the following above the `DocumentedContent` class:

```kotlin
/** Kotlin factory for documented content */
fun documentedContent(
        clazz: Class<*>,
        contentType: String? = ContentType.AUTODETECT,
        isArray: Boolean = false
): DocumentedContent {
    return DocumentedContent(
            clazz,
            contentType = contentType,
            isArray = isArray
    )
}

/** Kotlin factory for documented content */
fun documentedContent(
        oneOfClasses: OneOf,
        contentType: String? = ContentType.AUTODETECT,
        isArray: Boolean = false
): DocumentedContent {
    return DocumentedContent(
            oneOfClasses.classes.toList().toTypedArray(),
            contentType = contentType,
            isArray = isArray,
            schemaType = SchemaType.oneOf
    )
}

/** Kotlin factory for documented content */
fun documentedContent(
        anyOfClasses: AnyOf,
        contentType: String? = ContentType.AUTODETECT,
        isArray: Boolean = false
): DocumentedContent {
    return DocumentedContent(
            anyOfClasses.classes.toList().toTypedArray(),
            contentType = contentType,
            isArray = isArray,
            schemaType = SchemaType.anyOf
    )
}

/** Kotlin factory for documented content */
fun documentedContent(
        allOfClasses: AllOf,
        contentType: String? = ContentType.AUTODETECT,
        isArray: Boolean = false
): DocumentedContent {
    return DocumentedContent(
            allOfClasses.classes.toList().toTypedArray(),
            contentType = contentType,
            isArray = isArray,
            schemaType = SchemaType.anyOf
    )
}
```

Reply:
You can replace the test with

```kotlin
    @Test
    fun `createSchema() work with composed schemas`() {
        val app = Javalin.create {
            it.registerPlugin(OpenApiPlugin(OpenApiOptions(
                    Info().title("Example").version("1.0.0")
            )))
        }

        val anyOfDocumentation = document()
                .operation {
                    it.summary = "Get any of said object"
                    it.operationId = "anyOfRoute"
                }
                .result("200", documentedContent(anyOf(Address::class, User::class))) {
                    it.description = "returns any of the schemas"
                }
        app.get("/anyOfRoute", documented(anyOfDocumentation) {})

        val oneOfDocumentation = document()
                .operation {
                    it.summary = "Get one of said object"
                    it.operationId = "oneOfRoute"
                }
                .result("200", documentedContent(oneOf(Address::class, User::class))) {
                    it.description = "returns one of the schemas"
                }
        app.get("/oneOfRoute", documented(oneOfDocumentation) {})

        val allOfDocumentation = document()
                .operation {
                    it.summary = "Get all of said object"
                    it.operationId = "allOfRoute"
                }
                .result("200", documentedContent(allOf(Address::class, User::class))) {
                    it.description = "returns all of the schemas"
                }
        app.get("/allOfRoute", documented(allOfDocumentation) {})

        val actual = JavalinOpenApi.createSchema(app)
        assertThat(actual.asJsonString()).isEqualTo(oneOfAnyOfExampleJson.formatJson())
    }
```

Reply:
Then you can remove the other changes you made in `TestOpenApi.kt`